src/test/java/step4/CarTest.java

```diff
@@ -0,0 +1,39 @@
+package step4;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import step4.domain.Car;
+import step4.domain.Moving;
+import step4.domain.RandomMoving;
+import step4.view.result.CarMovementResult;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+
+public class CarTest {
+
+    @DisplayName("자동차는 5자 초과의 이름을 가질 수 없다.")
+    @Test
+    void constructor() {
+        assertThatIllegalArgumentException()
+                .isThrownBy(() -> Car.withName("다섯글자다다"));
+    }
+
+    @DisplayName("차량 움직임 테스트")
+    @Test
+    void move() {
+        final Car car = Car.withName("test");
+        final Moving strategy = new RandomMoving() {
+            @Override
+            public boolean movable() {
+                return true;
+            }
+        };
+
+        car.move(strategy);
```

Comment:
- 자동차가 이동하지 않는 경우에 대한 케이스도 추가해보면 어떨까요?
- 람다를 활용해보면 어떨까요?
```suggestion
        car.move(() -> true);
```